### DISCUSSION TITLE
Displaying a godot scene together with framelayout android (combine godot UI and android frameLayout view) have a problem

### DISCUSSION THREAD

**Discussion by Famouzy** [godotengine/godot-proposals#8909] [Platforms]
After creating a **google map** and displaying it in the **godot** by adding the view, there are only 2 options for displaying it - behind the scene and on top of it, which is the responsibility of the **shouldBeOnTop()** function
I tried different approaches, one of which is editing the export template and library for android studio (part of the code below, which adds getViewIndex, which is overridden in the android plugin and returns an integer). As a result, when changing the index, exactly the same thing happens as with the boolean **shouldBeOnTop**, i.e. below 2 is behind the scene, and 2 and above are on top of it.

The task is to combine applications with a ready UI on the yearot with framelayout android, to display additional menus and similar.

I would also add that when setting the index, let's say to 0 (**shouldBeOnTop - false**), setting the transparency of the main viewport has no effect on the display of the map, which is somewhere at the bottom, and just shows a black background.

**I will be glad to your suggestions. And, it is problem on this time? **

```
// Include the returned non-null views in the Godot view hierarchy.
for (plugin in pluginRegistry.allPlugins) {
	val pluginView = plugin.onMainCreate(activity)
	if (pluginView != null) {
		if (plugin.shouldBeOnTop()) {
			containerLayout?.addView(pluginView)
		} else {
			if (plugin.getViewIndex() >= 0) {
				containerLayout?.addView(pluginView, plugin.getViewIndex())
			}else{
				containerLayout?.addView(pluginView, 0)
			}
		}
	}
}
```

**Comment by AThousandShips** [comment]
This is for discussions about new features, please ask support questions on the other [community channels](https://godotengine.org/community)

**Reply by Famouzy** [reply]
Sorry. I assumed this might be a new feature in the future, since there might not be a way to implement what's in the discussion thread right now

**Reply by AThousandShips** [reply]
Then my bad sounded like asking for help with a problem and not a new feature 🙂

**Reply by Famouzy** [reply]
Yes, you are right, I corrected the text in places, again I apologize
